core/src/main/scala/package.scala

```diff
@@ -91,13 +92,10 @@ package object remotely {
     }
   }}
 
-  implicit val BitVectorMonoid = Monoid.instance[BitVector]((a,b) => a ++ b, BitVector.empty)
-  implicit val ByteVectorMonoid = Monoid.instance[ByteVector]((a,b) => a ++ b, ByteVector.empty)
-
   private[remotely] def fullyRead(s: Process[Task,BitVector]): Task[BitVector] = s.runFoldMap(x => x)
 
   private[remotely] def fixedNamedThreadPool(name: String): ExecutorService =
-    Executors.newFixedThreadPool(Runtime.getRuntime.availableProcessors, namedThreadFactory(name))
+    Executors.newFixedThreadPool(Runtime.getRuntime.availableProcessors.max(4), namedThreadFactory(name))
```

Comment:
They may not, it's true, and this isn't currently configurable. I'm going to remove this when I re-base as I don't believe it affects us (it was the worker threads going below 3 threads were we saw lags even when only sending one in at a time).  This is only currently used by endpoint when you have multiple endpoints set.